File: HybridAndroid/app/src/main/java/com/minkiapps/android/hybrid/ConnectionUtil.java
package com.minkiapps.android.hybrid;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.ServiceConnection;
import android.util.Log;

import com.huawei.ohos.localability.AbilityUtils;

public class ConnectionUtil {
    private static final String TAG = "ConnectionUtil";

    /**
     * 用于链接本地PA
     * @param context
     * @param connection
     */
    public static void connectAbility(Context context, String bundleName,
        String paName, ServiceConnection connection) {
        Intent aaIntent = new Intent();
        ComponentName component = new ComponentName(bundleName, paName);
        aaIntent.setComponent(component);
        try {
            AbilityUtils.connectAbility(context, aaIntent, connection);
        } catch (IllegalArgumentException e) {
            Log.e(TAG, "connectAbility: " + "abilility can't be found " + e.getMessage());
        }
    }

    /**
     * 断开PA链接
     * @param context
     * @param connection
     */
    public static void disconnectAbility(Context context, ServiceConnection connection) {
        AbilityUtils.disconnectAbility(context, connection);
    }

    public static void startAbility(Context context, String bundelName, String className) {
        Intent intent = new Intent();
        ComponentName cmp = new ComponentName(bundelName, className);
        intent.setComponent(cmp);
        AbilityUtils.startAbility(context, intent);
    }

    public static void startAbilityForResult(Context context, String bundelName,
        String className, int requestCode) {
        Intent intent = new Intent();
        ComponentName cmp = new ComponentName(bundelName, className);
        intent.setComponent(cmp);
        AbilityUtils.startAbilityForResult(context, intent, requestCode);
    }
}

File: HybridAndroid/app/src/main/java/com/minkiapps/android/hybrid/ConnectionUtil.kt
package com.minkiapps.android.hybrid

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.ServiceConnection
import android.os.Bundle
import com.huawei.ohos.localability.AbilityUtils
import timber.log.Timber

object ConnectionUtil {

    /**
     * 用于链接本地PA
     * @param context
     * @param connection
     */
    fun connectAbility(
        context: Context, bundleName: String,
        paName: String, connection: ServiceConnection
    ) {
        val aaIntent = Intent()
        val component = ComponentName(bundleName, paName)
        aaIntent.component = component
        try {
            AbilityUtils.connectAbility(context, aaIntent, connection)
        } catch (e: IllegalArgumentException) {
            Timber.e("ConnectAbility: Ability can't be found ${e.message}")
        }
    }

    /**
     * 断开PA链接
     * @param context
     * @param connection
     */
    fun disconnectAbility(context: Context, connection: ServiceConnection) {
        AbilityUtils.disconnectAbility(context, connection)
    }

    fun startAbility(context: Context,
                     bundleName: String,
                     className: String) {
        val intent = Intent()
        intent.putExtra("EXTRA_SOURCE", "ANDROID")
        val cmp = ComponentName(bundleName, className)
        intent.component = cmp
        AbilityUtils.startAbility(context, intent)
    }

    fun startAbilityForResult(
        context: Context, bundleName: String,
        className: String, requestCode: Int
    ) {
        val intent = Intent()
        val cmp = ComponentName(bundleName, className)
        intent.component = cmp
        AbilityUtils.startAbilityForResult(context, intent, requestCode)
    }
}